src/components/Mission.js

```diff
@@ -0,0 +1,107 @@
+import React, { useRef, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { missionButtonReducer } from '../redux/missions/MissionSlice';
+import getMission from '../redux/missions/MissionAPI';
+import '../styles/Mission.css';
+
+const Mission = ({
+  id,
+  mission,
+  description,
+}) => {
+  const member = useRef(null);
+  const buttonRef = useRef(null);
+  const dispatch = useDispatch();
+  const [missionText, setMissionText] = useState(
+    'Join Mission',
+  );
+  const [status, setStatus] = useState('NOT A MEMBER');
+  const { missionReserve } = useSelector(
+    (store) => store.mission,
+  );
+  useEffect(() => {
+    missionReserve.forEach((item) => {
+      if (item.id === id) {
+        setMissionText('Leave Mission');
+        setStatus('ACTIVE MEMBER');
+        member.current.classList.add('activeMember');
+        buttonRef.current.classList.add('leaveMission');
+      }
+    });
+  }, [id, missionReserve]);
+
+  function setMissionButton(e) {
+    if (missionText === 'Join Mission') {
+      setMissionText('Leave Mission');
+      setStatus('ACTIVE MEMBER');
+      member.current.classList.add('activeMember');
+      dispatch(
+        missionButtonReducer({ id, missionText, mission }),
+      );
+      e.target.classList.add('leaveMission');
+    } else {
+      setMissionText('Join Mission');
+      setStatus('NOT A MEMBER');
+      member.current.classList.remove('activeMember');
+      e.target.classList.remove('leaveMission');
+      dispatch(
+        missionButtonReducer({ id, missionText, mission }),
+      );
+    }
+  }
+
+  return (
+    <section className="missionWrapper">
+      <div className="data-row">
+        <h3>{mission}</h3>
+        <p>{description}</p>
+        <p ref={member} className="member">{status}</p>
+        <button
+          ref={buttonRef}
+          className="missionButton"
+          onClick={(e) => setMissionButton(e)}
+          type="button"
+        >
+          {missionText}
+        </button>
+      </div>
+    </section>
+  );
+};
+
+Mission.propTypes = {
+  id: PropTypes.string.isRequired,
+  mission: PropTypes.string.isRequired,
+  description: PropTypes.string.isRequired,
+};
+
+const MissionPage = () => {
+  const dispatch = useDispatch();
+  const { content } = useSelector((store) => store.mission);
+  useEffect(() => {
+    dispatch(getMission());
+  }, [dispatch]);
```

Comment:
The problem is whenever you go to missions it reFetch all the missions even when they are available in the state making unnecessary calls to API and burdening the server. 


## HOW TO DO THAT:--

You can achieve above by checking if missions are available in REDUX state or not and fetching if their length is zero:-

```
const Missions = () => {
  const { missions, isLoading, error } = useSelector((store) => store.missions);
  const dispatch = useDispatch();
  useEffect(() => {
    if (missions.length === 0) {
      dispatch(fetchMissions());
    }
  }, [dispatch, missions.length]);
```